src/main/java/duke/core/Parser.java

```diff
@@ -0,0 +1,241 @@
+        String minuteString = minute == 0 ? "" : ":" + minute;
+        if (hour == 0) {
+            return 12 + minuteString + "am";
+        } else if (hour == 12) {
+            return 12 + minuteString + "pm";
+        } else if (hour < 12) {
+            return hour + minuteString + "am";
+        } else {
+            return hour - 12 + minuteString + "pm";
+        }
+    }
+
+    private static String parseDateTime(String s) throws DukeException {
+        DateTimeFormatter f = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+        try {
+            LocalDateTime dateTime = LocalDateTime.parse(s, f);
+            int day = dateTime.getDayOfMonth();
+            String monthAllCaps = dateTime.getMonth().toString();
+            String monthFirstCap = monthAllCaps.substring(0, 1) + monthAllCaps.substring(1).toLowerCase();
+            int year = dateTime.getYear();
+            return intToOrdinal(day) + " of " + monthFirstCap + " " + year + ", " + timeToAmPm(dateTime);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Input date and time is not in dd/MM/yyyy HHmm format (24-hour clock) :-(");
+        }
+    }
+
+    /**
+     * Parses a <code>Task</code> from a string array.
+     *
+     * @param words The string array to be parsed.
+     * @return The <code>Task</code> that is parsed from the string array.
+     */
+    private static Task parseTask(String[] words) throws DukeException {
+        if (words[0].equals("todo")) {
+            return new ToDo(subString(words, 1, words.length));
+        } else if (words[0].equals("deadline")) {
+            int i = findIdx(words, "/by");
+            String description = subString(words, 1, i);
+            String by = subString(words, i + 1, words.length);
+            String dateTimeString = parseDateTime(by);
+            return new Deadline(description, dateTimeString);
+        } else {
+            assert words[0].equals("event") : "Instruction type should be event";
+            int i = findIdx(words, "/at");
+            String description = subString(words, 1, i);
+            String at = subString(words, i + 1, words.length);
+            String dateTimeString = parseDateTime(at);
+            return new Event(description, dateTimeString);
+        }
+    }
+
+    private static boolean isValidTaskId(String s) {
+        try {
+            int taskId = Integer.parseInt(s);
+            if (taskId < 1) {
+                return false;
+            } else {
+                return true;
+            }
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Checks for illegal user input and throws exceptions accordingly.
+     *
+     * @param words The string array to be checked.
+     */
+    private static void checkIllegalInstruction(String[] words) throws DukeException {
+        String fw = words[0];
+        boolean isDoneCommand = fw.equals("done");
```

Comment:
You could just use a massive switch statement to check for the first word, and have the default case return an "UnknownCommand" class that tells the user that his/her command is invalid